### PR TITLE
fix(tangle-dapp): Fix sort/scroll issues on nomination selection list modal

### LINKED ISSUE #2378: [BUG] Sort/scroll issues on set up nomination list modal
- [ ] Sort is not functional (see screen recording)
- [ ] When scrolling the list, items should remain hidden (see screenshot)

### Screen Recording
https://github.com/webb-tools/webb-dapp/assets/109539270/af7916fd-ddeb-4546-916a-dd6ed1ac7d20

Image

### PR DESCRIPTION
**Pull request by vutuanlinh2k2**
## Summary of changes

_Provide a detailed description of proposed changes._

- Fix sort/scroll issues on nomination selection list modal
- Fix overflow on nomination list modal

### Proposed area of change

_Put an `x` in the boxes that apply._

- [ ] `apps/bridge-dapp`
- [ ] `apps/hubble-stats`
- [ ] `apps/stats-dapp`
- [x] `apps/tangle-dapp`
- [ ] `apps/testnet-leaderboard`
- [ ] `apps/faucet`
- [ ] `apps/zk-explorer`
- [ ] `libs/webb-ui-components`

### Reference issue to close (if applicable)

_Specify any issues that can be closed from these changes (e.g. `Closes #233`)._

- Closes #2378 

### Screen Recording

_If possible provide a screen recording of proposed change._

https://github.com/webb-tools/webb-dapp/assets/69841784/b715317b-0bc8-4ee0-b4ba-ffe48edf7897

---

### Code Checklist

_Please be sure to add .stories documentation if any additions are made to `libs/webb-ui-components`._

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: apps/tangle-dapp/components/ValidatorSelectionTable/ValidatorSelectionTable.tsx [hunk 1]
```diff
@@ -123,84 +115,49 @@ const ValidatorSelectionTable: FC<ValidatorSelectionTableProps> = ({
...
-          if (rowASelected && !rowBSelected) {
-            return -1;
-          }
-
-          if (!rowASelected && rowBSelected) {
-            return 1;
-          }
-
-          return 0;
-        },
```
Comment (AtelyPham) [inline apps/tangle-dapp/components/ValidatorSelectionTable/ValidatorSelectionTable.tsx:145]: I believe we should first keep the selected validators sorted.

File: apps/tangle-dapp/components/ValidatorSelectionTable/ValidatorSelectionTable.tsx [hunk 2]
```diff
@@ -214,43 +171,28 @@ const ValidatorSelectionTable: FC<ValidatorSelectionTableProps> = ({
...
-            } else if (newSorting[0].id === 'address') {
-              return newSorting;
-            } else {
-              return [SELECTED_VALIDATORS_COLUMN_SORT, ...newSorting];
-            }
-          });
-        } else {
-          setSorting(updaterOrValue);
-        }
-      },
```
Comment (AtelyPham) [inline apps/tangle-dapp/components/ValidatorSelectionTable/ValidatorSelectionTable.tsx:253]: Likewise, the logic for prioritizing selected validators in the sorting process must remain intact.

File: apps/tangle-dapp/components/ValidatorSelectionTable/ValidatorSelectionTable.tsx [hunk 3]
```diff
@@ -123,84 +115,49 @@ const ValidatorSelectionTable: FC<ValidatorSelectionTableProps> = ({
...
-          <div
-            className="flex items-center justify-center cursor-pointer"
-            onClick={header.column.getToggleSortingHandler()}
-          >
-            <HeaderCell title="Total Staked" className="flex-none block" />
-
-            <SortArrow column={header.column} />
-          </div>
+        header: () => (
+          <HeaderCell title="Total Staked" className="justify-start" />
```
Comment (AtelyPham) [inline apps/tangle-dapp/components/ValidatorSelectionTable/ValidatorSelectionTable.tsx:128]: It appears that you have not passed the 'toggleSortingHandler' into this component to enable sorting toggling.

**Review by AtelyPham — COMMENTED** [review]
Left some comments.

File: apps/tangle-dapp/components/ValidatorSelectionTable/ValidatorSelectionTable.tsx (hunk 3 above)
Comment (vutuanlinh2k2) [reply apps/tangle-dapp/components/ValidatorSelectionTable/ValidatorSelectionTable.tsx:128]: The `Total Staked` column is sortable as shown the video

**Review by monaiuu — APPROVED** [review]

**Comment by vutuanlinh2k2** [comment]
@AtelyPham I updated the logic to make sure all selected validators always stay on top no matter what field are being sorted

https://github.com/webb-tools/webb-dapp/assets/69841784/4b15abc7-eb1a-43bd-9974-fbea3da1a62a

**Comment by vutuanlinh2k2** [comment]
> Previously, when users select the validator, it should also be sorted to the top.
> 
>  CleanShot.2024-07-11.at.19.55.26.mp4

Fixed with the new commit

**Pull request merged**